apollo-router/src/plugins/telemetry/tracing/jaeger.rs

```diff
@@ -27,65 +25,55 @@ use crate::plugins::telemetry::tracing::SpanProcessorExt;
 use crate::plugins::telemetry::tracing::TracingConfigurator;
 
 #[derive(Debug, Clone, Deserialize, Serialize, JsonSchema)]
-// Can't use #[serde(deny_unknown_fields)] because we're using flatten for endpoint
-pub(crate) struct Config {
-    #[serde(flatten)]
-    #[schemars(schema_with = "endpoint_schema")]
-    pub(crate) endpoint: Endpoint,
-
-    #[serde(default)]
-    pub(crate) batch_processor: BatchProcessorConfig,
-}
-
-// This is needed because of the use of flatten.
-fn endpoint_schema(gen: &mut SchemaGenerator) -> Schema {
-    let mut schema: SchemaObject = <Endpoint>::json_schema(gen).into();
+#[serde(deny_unknown_fields, untagged)]
+pub(crate) enum Config {
```

Comment:
why is the configuration shape changing here?

Reply:
It's a restructuring that doesn't change the shape for the user. 

Previously there was some manual schema generation and unfortunate use of flatten, this also meant that the auto doc generation didn't work.

The refactoring here uses an enum with untagged to give the same format but without all of the hackery that was there previously.

Reply:
alright, good!